osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -270,43 +363,127 @@ void testHappyWithDifferentTimeZones(final String timeZone) throws Exception {
+
+    if (areDatesEqual(periodicMeterReads.get(0).getLogTime(), timeMeterRead1)) {
+      this.checkValues(
+          periodicMeterReads.get(0),
+          timeMeterRead1,
+          meterValueImport11,
+          meterValueImport12,
+          meterValueExport11,
+          meterValueExport12);
+      this.checkValues(
+          periodicMeterReads.get(1),
+          timeMeterRead2,
+          meterValueImport21,
+          meterValueImport22,
+          meterValueExport21,
+          meterValueExport22);
+    } else {
+      {
+        this.checkValues(
+            periodicMeterReads.get(1),
+            timeMeterRead1,
+            meterValueImport11,
+            meterValueImport12,
+            meterValueExport11,
+            meterValueExport12);
+        this.checkValues(
+            periodicMeterReads.get(0),
+            timeMeterRead2,
+            meterValueImport21,
+            meterValueImport22,
+            meterValueExport21,
+            meterValueExport22);
+      }
+    }
+  }
 
-    periodicMeterReads.forEach(p -> assertThat(p.getLogTime()).isNotNull());
+  private void checkValues(
+      final PeriodicMeterReadsResponseItemDto response,
+      final CosemDateTimeDto time,
+      final DlmsMeterValueDto import1,
+      final DlmsMeterValueDto import2,
+      final DlmsMeterValueDto export1,
+      final DlmsMeterValueDto export2) {
+    assertThat(areDatesEqual(response.getLogTime(), time)).isTrue();
+    assertThat(response.getActiveEnergyImportTariffOne()).isEqualTo(import1);
+    assertThat(response.getActiveEnergyImportTariffTwo()).isEqualTo(import2);
+    assertThat(response.getActiveEnergyExportTariffOne()).isEqualTo(export1);
+    assertThat(response.getActiveEnergyExportTariffTwo()).isEqualTo(export2);
   }
 
-  private AttributeAddressForProfile createAttributeAddressForProfile(
-      final DlmsObject dlmsObject, final List<DlmsCaptureObject> selectedObjects) {
-    return new AttributeAddressForProfile(
-        new AttributeAddress(
-            dlmsObject.getClassId(),
-            dlmsObject.getObisCode(),
-            dlmsObject.getDefaultAttributeId(),
-            null),
-        selectedObjects);
+  private DlmsDevice createDevice(final Protocol protocol) {
+    final DlmsDevice dlmsDevice = new DlmsDevice();
+    dlmsDevice.setProtocol(protocol);
+    dlmsDevice.setSelectiveAccessPeriodicMeterReadsSupported(true);
+    return dlmsDevice;
   }
 
-  private AttributeAddress createAttributeAddress(final DlmsObject dlmsObject) {
-    return new AttributeAddress(
-        dlmsObject.getClassId(), dlmsObject.getObisCode(), dlmsObject.getDefaultAttributeId());
+  private ProfileGeneric createProfile() {
+    final Attribute attributeCaptureObjects =
+        this.createAttribute(
+            3,
+            "CLOCK,2|AMR_PROFILE_STATUS,2|ACTIVE_ENERGY_IMPORT_RATE_1,2|ACTIVE_ENERGY_IMPORT_RATE_2,2|ACTIVE_ENERGY_EXPORT_RATE_1,2|ACTIVE_ENERGY_EXPORT_RATE_1,2|MBUS_MASTER_VALUE,2|MBUS_MASTER_VALUE,5");
```

Comment:
ACTIVE_ENERGY_EXPORT_RATE_1,2 twice. Should be ACTIVE_ENERGY_EXPORT_RATE_**2**,2